tests/algorithms/pytorch/test_newton_raphson.py

```diff
@@ -206,6 +210,27 @@ def forward(self, x):
         my_algo.train(datasamples=(x_train, y_train), _skip=True)
 
 
+@pytest.mark.parametrize(
+    "n_samples,y_shape",
+    [(5, 3), (5, 1), (10, 2), (1, 1), (1, 10)],
+)
+def test_prediction_shape(session_dir, perceptron, torch_algo, n_samples, y_shape):
+    x_shape = 15
+
+    x_train = np.zeros([n_samples, x_shape])
+    y_train = np.ones([n_samples, y_shape])
+
+    model = perceptron(linear_n_col=x_shape, linear_n_target=y_shape)
+    my_algo = torch_algo(model=model, batch_size=1)
+
+    my_algo.train(datasamples=(x_train, y_train), _skip=True)
+    prediction_folder = Path(tempfile.mkdtemp(dir=session_dir))
+    my_algo.predict(datasamples=(x_train, y_train), predictions_path=prediction_folder, _skip=True)
+    file_name = str(prediction_folder).split("/")[-1]
+    prediction = np.load(os.path.join(session_dir, file_name, file_name + ".npy"))
```

Comment:
With the change on `prediction_path`, you should just do `np.load(prediction_path)`